src/coreclr/nativeaot/Runtime/unix/UnwindHelpers.cpp

```diff
@@ -834,32 +834,97 @@ bool DoTheStep(uintptr_t pc, UnwindInfoSections uwInfoSections, REGDISPLAY *regs
 #endif
 
 #elif defined(_LIBUNWIND_ARM_EHABI)
-    uc.setInfoBasedOnIPRegister(true);
+    // If there is ARM EHABI unwind info, look there next.
+    if (uwInfoSections.arm_section == 0 || !this->getInfoFromEHABISection(pc, uwInfoSections))
```

Comment:
There's notable difference in behaviour in this code path. It basically inlines the `UnwindCursor::setInfoBasedOnIPRegister` API for `_LIBUNWIND_ARM_EHABI` case.

However, the previous code set the `isReturnAddress` parameter to `true` which took this code path:
https://github.com/dotnet/runtime/blob/a64420c79cb63c485138f4f1352b8730f27d7b19/src/native/external/llvm-libunwind/src/UnwindCursor.hpp#L1969-L1975

I am not sure whether it should adjust the PC pointer or not. I don't think the EHABI code path is currently tested, and the DWARF code path above it doesn't make the same adjustment.